src/platform/checksum.cpp

```diff
@@ -0,0 +1,37 @@
+#include "checksum.h"
+#include "filesystem.h"
+#include "ifile.h"
+#include <fstream>
+
+namespace platform {
+    boost::uint32_t fileChecksum( std::string const& filename )
+    {
+        boost::uint32_t checksum(0);
```

Comment:
on the server side it's calculated only once for each file (saved in static vars), on the client side it's supposed to receive once for each connecting.
do you see use cases when it maybe calculated several times?